Mathlib/Data/Finite/Basic.lean

```diff
@@ -140,10 +135,11 @@ instance Quotient.finite {α : Sort _} [Finite α] (s : Setoid α) : Finite (Quo
   Quot.finite _
 #align quotient.finite Quotient.finite
 
-instance Function.Embedding.finite {α β : Sort _} [Finite β] : Finite (α ↪ β) :=
-  by
+instance Function.Embedding.finite {α β : Sort _} [Finite β] : Finite (α ↪ β) := by
   cases' isEmpty_or_nonempty (α ↪ β) with _ h
-  · infer_instance
+  · -- Porting note: inferInstance fails because it applies `Finite.of_fintype`.
```

Comment:
Why does this make it fail?

Reply:
```suggestion
  · -- Porting note: inferInstance fails because it applies `Finite.of_fintype` and produces a
    -- "stuck at solving universe constraint" error.
```

Reply:
It's related to the MWE that Reid reported here: https://leanprover.zulipchat.com/#narrow/stream/341532-lean4-dev/topic/synthInstances.20.26.20universe.20metavariables